### PR TITLE
[DOCS] Adds top_metrics aggs examples to Transform docs

### PR DESCRIPTION
**Pull request by szabosteve**
## Overview

This PR adds two new examples to the Transform examples. Both of them show how to use `top_metrics` in transforms.

### Preview

* [Bytes example](https://elasticsearch_72738.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/transform-examples.html#example-bytes)
* [Email and name example](https://elasticsearch_72738.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/transform-examples.html#example-customer-names)

### CODE REVIEW

File: docs/reference/transform/examples.asciidoc [hunk 1]
```diff
@@ -16,6 +16,9 @@ step-by-step example, see <<ecommerce-transforms>>.
 * <<example-clientips>>
 * <<example-last-log>>
```
Comment (szabosteve) [inline docs/reference/transform/examples.asciidoc:18]:
```suggestion
```

File: docs/reference/transform/examples.asciidoc [hunk 2]
```diff
@@ -483,3 +487,206 @@ The search result shows you data like this for each client IP:
...
+      "last" : {
+        "customer_last_name.keyword" : "Long",
+        "customer_first_name.keyword" : "Recip",
+        "email" : "recip@long-family.zzz"
+      },
+      "customer_id" : "10"
+    },
+    {
+      "last" : {
+        "customer_last_name.keyword" : "Jackson",
```
Comment (hendrikmuhs) [inline docs/reference/transform/examples.asciidoc:674]: FYI I opened #72792 to discuss if we should remove `.keyword` from the output field name.

**Review by hendrikmuhs — APPROVED** [review]
LGTM

File: docs/reference/transform/examples.asciidoc (hunk 2 above)
Comment (szabosteve) [reply docs/reference/transform/examples.asciidoc:674]: Thanks, @hendrikmuhs I subscribed to that issue, so I can come back to the examples and amend them according to the outcome of the discussion.

File: docs/reference/transform/examples.asciidoc [hunk 3]
```diff
@@ -483,3 +487,206 @@ The search result shows you data like this for each client IP:
...
+{transform} with a <<search-aggregations-metrics-top-metrics,`top_metrics`>> 
+aggregation.
+
+Group the data by a <<_date_histogram,date histogram>> on the time field with an 
+interval of one hour. Use a 
+<<search-aggregations-metrics-max-aggregation,max aggregation>> on the `bytes` 
+field to get the maximum amount of data that is sent to the server. Without 
+the `max` aggregation, the API call still returns the client IP that sent the 
+most bytes, however, the amount of bytes that it sent is not returned. In the 
+`top_metrics` property, specifiy `clientip` and `geo.src`, then sort them by the
```
Comment (lcawl) [inline docs/reference/transform/examples.asciidoc:506]:
```suggestion
`top_metrics` property, specify `clientip` and `geo.src`, then sort them by the 
```

File: docs/reference/transform/examples.asciidoc [hunk 4]
```diff
@@ -483,3 +487,206 @@ The search result shows you data like this for each client IP:
...
+          }
+        }
+      }
+    }
+  }
+}
+----------------------------------
+// TEST[skip:setup kibana sample data]
+
+<1> The data is grouped by a date histogram of the time filed with a one hour
```
Comment (lcawl) [inline docs/reference/transform/examples.asciidoc:554]:
```suggestion
<1> The data is grouped by a date histogram of the time field with a one hour 
```

**Review by lcawl — APPROVED** [review]
Two minor comments, otherwise LGTM

**Pull request merged**